src/numericalAnalysis.cpp

```diff
@@ -0,0 +1,119 @@
+#include "mantella_bits/numericalAnalysis.hpp"
+
+// C++ standard library
+#include <cmath>
+#include <limits>
+
+// Mantella
+#include "mantella_bits/assert.hpp"
+
+namespace mant {
+  double brent(
+      std::function<double(double)> objectiveFunction,
+      double lowerBound,
+      double upperBound,
+      double xTolerance) {
+    double lowerBoundObjectiveValue = objectiveFunction(lowerBound);
```

Comment:
In addition to the comment above, we also have the special case that both are equal. In this case, both must either be close enough to the root, or it is known to be impossible to find any root in this interval, which should lead to an exception.

```
if (lowerBound == upperBound) {
  verify(lowerBoundObjectiveValue < 1e-12, "brent: Both bounds are equal, but neither is close enough to the root, resulting in an impossible to solve problem.");
}
```